.github/workflows/builder_docker-based_slsa3.yml

```diff
@@ -0,0 +1,191 @@
+
+# TODO: env
+
+defaults:
+  run:
+    shell: bash
+
+###################################################################
+#                                                                 #
+#            Input and output argument definitions                #
+#                                                                 #
+###################################################################
+on:
+  workflow_call:
+    inputs:
+      builder-image:
+        description: >
+          The OCI image name of the builder image responsible for 
+          creating the artifact. This must not include a tag or digest.
+        required: true 
+        type: string
+      builder-digest:
+        description: >
+          The OCI image digest of the builder-image. 
+          This must be specified in order to fully specify the builder 
+          image. The image digest is of the form '<alg>:<digest>'
+        required: true 
+        type: string
+      config:
+        description: >
+          Path to a configuration file containing a command that the builder
+          image should be invoked with and optionally, docker options
+          and environment variables.
+        required: true 
+        type: string
+      output:
+        description: >
+          Path to the output artifact(s) relative to the root of the Git
+          repository.
+          Users may specify a single artifact or use a shell-style wildcard
+          pattern to match multiple artifacts.
+        required: true 
+        type: string
+      source-repository:
+        description: >
+          Source repository to checkout for the build. For example, octocat/hello-world.
+          The builder will mount the root of this source repository to the workspace.
+
+          By default, uses the source repository of the caller workflow.
+        required: false 
+        type: string
+      source-digest:
+        description: >
+          The SHA of the source repository to checkout. 
+
+          By default, uses the source repository of the caller workflow.
+        required: false
+        type: string
+      # TODO: Options for requesting and verifying provenance of builder image
+      # This will use slsa-verifier and slsa-verifier option.
+      #      * request-builder-verification
+
+      # TODO: Authentication options for the builder-image.
+      #      * registry-username
+      #      * registry-password
+
+jobs:
+  ###################################################################
+  #                                                                 #
+  #           Optional: Verify builder-image provenance             #
+  #                                                                 #
+  ###################################################################
+  verify-builder:
+    # This is a no-op right now.
+    outputs:
+      verified: ${{ steps.verify.outputs.verified }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: run no-op slsa-verifier
```

Comment:
Will the container download require authentication? I think it might, we we may need secret in workflow arguments.

Think about how we can express this in the BYOB Option 1. At the moment we only support binaries to download. We can either add an option for container images in the BYOB config file, or let the "callback Action" pull the image. (Probably a better option if secrets are necessary)

Reply:
The container download will happen when performing the build, so the `build` step may require authentication. I'll leave a TODO there. For now, Oak's is public, but when we test we can add in secret workflow arguments (also labeled TODO)